### ISSUE TITLE
Analyze branch checkout speed

### ISSUE THREAD

**Issue by sorincarbunaru** [oxygenxml/oxygen-git-plugin#113]
I find the git client rather slow when switching between branches as well. I'm not sure how to speed up the screen display after loading a new local branch.

_Originally posted by @codemaestro in https://github.com/oxygenxml/oxygen-git-plugin/issues/42#issuecomment-650890978_

**Comment by sorincarbunaru** [comment]
@codemaestro, is this problem still present in our Git Client? BTW, our latest version is 2.3.1. If the problem is still there, could you please tell us just how long does switching between branches take? And could you compare on you machine how long the same scenario takes in other Git clients such as Source Tree, GitHub Desktop or Git Kraken?

**Comment by codemaestro** [comment]
@sorincarbunaru, the latest plugin version seems feels about the same speed, but considering all the validation that Oxygen is probably performing in the backend it should be acceptable. When I switch to a local/remote branch, I saw it takes about 5 seconds for the plugin to register that the branch checkout was complete. This was when using both the Git Staging panel (for local branches) and the Git Branch panel (for local AND remote branches).

I don't actually use another git client except the command line, and VS Code's git support. So I timed the same type of change using VS Code, and branch changes take about 2 seconds to finalize the checkout. But there isn't any other validation activity that could hold up the checkout message.

I hope this helps.
